File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnTertiaryDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnTertiaryDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd34
            AppTheme.STANDARD -> R.color.cstd34
            AppTheme.BLUE -> R.color.cbd34
            AppTheme.GREEN -> R.color.cgd34
            AppTheme.RED -> R.color.crd34
            AppTheme.NOTHING -> R.color.cnod34
        }
    }
}

File: lib/layer2_core/dark/e_on_tertiary_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnTertiaryDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd34;
      case 'standard': return cstd34;
      case 'blue': return cbd34;
      case 'green': return cgd34;
      case 'red': return crd34;
      case 'nothing': return cnod34;
      default: return cstd34;
    }
  }
}
